src/librustdoc/json/conversions.rs

```diff
@@ -562,16 +554,28 @@ impl FromWithTcx<clean::Trait> for Trait {
             is_unsafe,
             items: ids(items, tcx),
             generics: generics.into_tcx(tcx),
-            bounds: bounds.into_iter().map(|x| x.into_tcx(tcx)).collect(),
+            bounds: bounds.into_tcx(tcx),
             implementations: Vec::new(), // Added in JsonRenderer::item
         }
     }
 }
 
-impl FromWithTcx<Box<clean::Impl>> for Impl {
-    fn from_tcx(impl_: Box<clean::Impl>, tcx: TyCtxt<'_>) -> Self {
+impl FromWithTcx<clean::PolyTrait> for PolyTrait {
+    fn from_tcx(
+        clean::PolyTrait { trait_, generic_params }: clean::PolyTrait,
+        tcx: TyCtxt<'_>,
+    ) -> Self {
+        PolyTrait {
+            trait_: clean::Type::Path { path: trait_ }.into_tcx(tcx),
```

Comment:
Ideally we keep it in Path form rather than allowing it to be any type. (Type definition needs to be updated too.)

```suggestion
            trait_: trait_.into_tcx(tcx),
```

Reply:
We don't yet have a `ResolvedPath` type in the json output, so I think that should be done in a followup change, as theirs a couple of other places we'd also want to use it (https://github.com/rust-lang/rust/blob/e141246cbbce2a6001f3181d3d0f661bbfd9c7ea/src/librustdoc/json/conversions.rs#L423, https://github.com/rust-lang/rust/blob/e141246cbbce2a6001f3181d3d0f661bbfd9c7ea/src/librustdoc/json/conversions.rs#L500)

Reply:
Ok, makes sense to hold off then.